libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -266,6 +267,67 @@ std::string YulUtilFunctions::requireOrAssertFunction(bool _assert, Type const*
 	});
 }
 
+std::string YulUtilFunctions::requireWithErrorFunction(FunctionCall const& errorConstructorCall)
+{
+	ErrorDefinition const* errorDefinition = dynamic_cast<ErrorDefinition const*>(ASTNode::referencedDeclaration(errorConstructorCall.expression()));
+	solAssert(errorDefinition);
+
+	std::string const errorSignature = errorDefinition->functionType(true)->externalSignature();
+	std::string const signatureHash = formatNumber(util::selectorFromSignatureU256(errorSignature));
+	std::string const functionName = [&]() {
+		// Note that in most cases we'll always generate one function per error definition,
+		// because types in the constructor call will match the ones in the definition. The only
+		// exception are calls with types, where each instance has its own type (e.g. literals).
+		std::string name = "require_helper_t_error_" + std::to_string(errorDefinition->id()) + "_" + errorDefinition->name();
+		for (ASTPointer<Expression const> const& argument: errorConstructorCall.sortedArguments())
+		{
+			solAssert(argument->annotation().type);
+			name += ("_" + argument->annotation().type->identifier());
+		}
+		return name;
+	}();
+
+	auto errorArgumentVarsAndTypes = [&]() -> std::tuple<std::vector<std::string>, std::vector<Type const*>> {
```

Comment:
With `auto` it looks like value being assigned is the lambda at first. Would be much more readable this way:
```suggestion
	std::tuple<std::vector<std::string>, std::vector<Type const*>> errorArgumentVarsAndTypes = [&]() {
```

Reply:
Yeah, not sure why I did that :)